### DISCUSSION TITLE
Android端是不可用使用ipv6吗

### DISCUSSION THREAD

**Discussion by Daniel19491001** [wh201906/SerialTest#34] [Q&A] [answered]
我看电脑端可以检测到ipv6的地址，但是手机端检测不到，我是只开流量，然后设置成仅ipv6地址，但是还是没办法显示ipv6地址

**Comment by wh201906** [comment]
这个应该是Qt库的问题，在Android上无法~~拿到~~显示本机的IPv6地址，但实际上IPv6功能是可用的
TCP客户端模式下本机地址那里无法显示IPv6地址，但是远端地址填IPv6地址可以连上TCP服务器
TCP服务器模式下如果本机地址选“(任意)”也可以接受IPv6连接

**Reply by Daniel19491001** [reply]
Qt库这个问题我这边可以解决吗，我现在是准备电脑和手机都走公网ipv6，ipv4没有公网IP，所以现在是连接不上

**Reply by wh201906** [reply]
等我找个有IPv6的公网环境测试下

**Reply by wh201906** [reply, marked as answer]
我测下来是可以连接的，手机或者电脑做TCP服务器都可以。
测试条件是电脑过路由器连公网，手机走流量连公网，两边都有IPv6公网地址。你只要知道作为TCP服务器一端的IPv6地址就可以连接了
你可以访问 http://www.test-ipv6.com/ 来查询地址

**Reply by Daniel19491001** [reply]
感谢帮忙测试，我现在是用zerotier调通了，舍近求远了，我之后找个有公网ipv6的电脑试一下，应该没问题

**Reply by wh201906** [reply]
我又加了点代码
https://github.com/wh201906/SerialTest/commit/03a0e7938c931e60fd66865ed66706920fd37322
下一个版本在Android上也能显示本机IPv6地址了

**Reply by wh201906** [reply]
新版本可以显示IPv6地址了，你可以试一下
https://github.com/wh201906/SerialTest/releases/tag/V0.3.3

**Comment by wh201906** [comment]
如果你的路由器支持IPv6，那么你的Android设备在局域网下也可以走IPv6连接
